packages/web3-eth-contract/src/contract.ts

```diff
@@ -1076,12 +1077,23 @@ export class Contract<Abi extends ContractAbi>
 			input: undefined,
 			from: modifiedContractOptions.from ?? this.defaultAccount ?? undefined,
 		};
-		const tx = getSendTxParams({
-			abi,
-			params,
-			options: { ...options, dataInputFill: this.config.contractDataInputFill },
-			contractOptions: modifiedContractOptions,
-		});
+		let tx;
+		if (!isNullish(this.currentProvider) && isMetaMaskProvider(this.currentProvider)) {
```

Comment:
This will not fix if some one is using directly sendTransaction via `web3.eth` package and specifying some `data`. 

As this is not good but only temporary fix until issue is fixed in metamask ( so that metamask dnt discard input in sendTransaction ) , make changes in https://github.com/web3/web3.js/blob/f860b0481d7c1ef09ddaeb33098b2253ca694150/packages/web3-core/src/web3_request_manager.ts#L177 :

detect if provider is metamask and request method is `eth_sendTransaction`, send `data` instead of `input`

Reply:
also add code comments having metamask issue URL